lib/Fmt.mli

```diff
@@ -87,10 +101,6 @@ val str_as : int -> string -> t
 val opt : 'a option -> ('a -> t) -> t
 (** Format an option using provided formatter for the element. *)
 
-val list : 'a list -> s -> ('a -> t) -> t
```

Comment:
We could have `val list : 'a list -> sp -> ('a -> t) -> t`, I think most separators are actually spaces anyway

Reply:
I was quite annoyed in the past with the limited `list` function. I'd suggest instead that we remove all the `_k` in the function names and have only functions taking a `t`.

I think the extra work to pass just a space is worth the consistency: one way to build a space, one way to build a list, the list can evolve in a more complex form later.

Reply:
Another option is to have an infix operator for `str`. This could remove some of the parenthesis if the operator has higher priority than str. It would otherwise make the code a bit higher-context : `str` is self-explanatory, `!$` is not.

Overall, I think that `!$` combined with values for common strings like `"\n"` and `" "` is my preferred option. It also has the advantage of being likely easier to implement with regexp, but I am not completely against `wrap_str` either.

Reply:
I don't mind the many `str " "` for now and I think we should stay focused on one change at a time.

I like the idea of the prefix operator. I don't think we need a self-explanatory name for the most important abstraction in the project, especially if the `Fmt` module is not too big.
Again, for later.

Reply:
I am happy for this kind of changes to be in an other PR. If @gpetiot is convinced as well I think we can merge.

Reply:
Yes it's all good to me!